packages/tupaia-web-server/src/routes/EntitiesRoute.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import { Request } from 'express';
+import { Route } from '@tupaia/server-boilerplate';
+import { Entity } from '@tupaia/types';
+
+export type EntitiesRequest = Request<
+  any,
+  any,
+  any,
+  any
+>;
+
+interface NestedEntity extends Entity {
+  parent_code?: string | null;
+  children?: NestedEntity[] | null;
+}
+
+export class EntitiesRoute extends Route<EntitiesRequest> {
+  public async buildResponse() {
+    const { params, query, ctx } = this.req;
+    const { rootEntityCode, hierarchyName } = params;
+
+    // TODO: Allow filtering by generational distance in entity server
+    const flatEntities = await ctx.services.entity.getDescendantsOfEntity(hierarchyName, rootEntityCode, {}, true);
```

Comment:
Currently `entity-server` doesn't have a way to filter by generational distance (i.e. return only X layers of hierarchy), so this always fetches the full list of children which can be quite large.

I'll split that change to a new PR

Reply:
@EMcQ-BES seems like @kael89 timed this perfectly. The generational_distance filter was added recently, here: https://github.com/beyondessential/tupaia/pull/4638

Reply:
Updated this to add a default generational_distance filter 😎